src/main/java/org/team1540/base/power/PowerManager.java

```diff
@@ -52,19 +65,53 @@ public void run() {
     while (true) {
       // No whiles in here as that'd stop the last block from executing
       if (running) {
-        if (isSpiking()) {
-          if (theTimer.get() <= 0) {
-            // Calling the timer when it's already started seems to reset it.
-            theTimer.start();
+
+        // Zach says don't break out into method. I don't care that much.
+
+        boolean stopScaling = true;
+
+        if (isCurrentSpiking()) {
+          if (isCurrentSpiking()) {
+            if (currentTimer.get() <= 0) {
+              // Calling the timer when it's already started seems to reset it.
+              currentTimer.start();
+            }
+            if (hasTimePassedCurrent()) {
+              scalePower();
+            }
+          } else {
+            currentTimer.stop();
+            currentTimer.reset();
           }
-          if (timeHasPassed()) {
-            scalePower();
+          stopScaling = false;
+        } else {
+          currentTimer.stop();
+          currentTimer.reset();
+        }
+
+        if (isVoltageDipping()) {
+          if (isVoltageDipping()) {
+            if (voltageTimer.get() <= 0) {
+              // Calling the timer when it's already started seems to reset it.
+              voltageTimer.start();
+            }
+            if (hasTimePassedVoltage()) {
```

Comment:
Could you not break this out into a separate if statement with `hasTimePassedVoltage() || hasTimePassedCurrent()`? Might require variable extraction but still